### PR TITLE
Restructure user search section

### PR DESCRIPTION
**Pull request by erjohnson**

### CODE REVIEW

**Comment by siacomuzzi** [comment]
please review/merge this too: https://github.com/auth0/api2/pull/2299

**Review by mpaktiti — CHANGES_REQUESTED** [review]
If I go to https://auth0-docs-content-pr-5900.herokuapp.com/docs/users/search (using the menu User Management > User Search) it throws an error.
(versioning bug, you need to add a specific version to the sidebar URL)

Also, this is under Articles without breadcrumbs
https://auth0-docs-content-pr-5900.herokuapp.com/docs/users/search/v3
and I use the versioning dropdown to go to v2, it is under APIs https://auth0-docs-content-pr-5900.herokuapp.com/docs/users/search/v2

Can you fix those and check the review app that all work ok? It should all be under Articles, with breadcrumbs, and working menus.

**Comment by erjohnson** [comment]
In my local build, switching versions switches the sections from users management to api v2?

**Comment by mpaktiti** [comment]
@erjohnson It does change versions but it also changes tabs (v3 under Articles, v2 under APIs). Probably because the v2 doc has still the metadata `section: apis`, if you remove it it should work.

**Comment by mpaktiti** [comment]
@erjohnson One more thing, if I click on `Query Syntax` (articles sidebar menu: User Management > User Search > Query Syntax) I get a 404 
https://auth0-docs-content-pr-5900.herokuapp.com/docs/users/search/query-sytax
(probably because you haven't sent the version as part of the URL at the sidebar config file)

File: articles/users/search/v2/index.md [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ---
 description: This page lists several examples of user search queries using query string syntax.
-section: apis
 crews: crew-2
 toc: true
 ---
```
Comment (mpaktiti) [inline articles/users/search/v2/index.md:5]: Just a note that we do not have a version-warning panel here. Not sure if we want to that's why I'm not asking for the change.

File: articles/users/search/v3/index.md [hunk 2]
```diff
@@ -289,5 +176,3 @@ Search by date | `updated_at:<2018-01-15` | `updated_at:[* TO 2018-01-15}`
 Search by date | `last_login:<=2017-12` | `last_login:[* TO 2017-12]`
 String exact match | `name.raw:"john richard doe"` | `name:"john richard doe"`
 Phrase contains a word | `name:"richard"`, `name:richard` | `name:*richard*`
-
```
Comment (mpaktiti) [inline articles/users/search/v3/index.md]: Can you add a next-steps element at the end of this doc, listing these two?
articles/users/search/v3/query-syntax.md
articles/users/search/v3/best-practices.md

Add it on its own section, like `Keep reading`

File: config/redirects.js [hunk 3]
```diff
@@ -1683,11 +1683,19 @@ module.exports = [
...
   },
   {
-    from: `/guides/login/migration-embedded-centralized`,
-    to: `/guides/login/migration-embedded-universal`
+    from: '/guides/login/migration-embedded-centralized',
+    to: '/guides/login/migration-embedded-universal'
+  },
+  {
+    from: '/api/management/v2/user-search',
+    to: '/users/search'
```
Comment (mpaktiti) [inline config/redirects.js]: Since these two do not have the `v2` in the URL, they will redirect to the default one, which is v3. Is this intentional?

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Looks good Eric, just a couple of items to address and it's good to go.

File: articles/users/search/v2/index.md [hunk 4]
```diff
@@ -6,6 +6,10 @@ toc: true
 
 # User Search
 
+::: version-warning
+This document covers a previous version of user search. We recommend you use <a href="/users/search/v3">user search v3</a>.
```
Comment (mpaktiti) [inline articles/users/search/v2/index.md]: @erjohnson For future reference, you don't have to use HTML within the UI elements, you can use Markdown and the result is the same. I'm going to fix these two for you and merge.

**Review by mpaktiti — APPROVED** [review]
Thanks for all the changes @erjohnson. This looks good so I'm gonna go ahead and merge. Two notes on some changes I pushed:
- In the version-warning panels for the v2 docs, you used HTML for the links. I replaced it with Markdown. This stand for all UI elements, you don't have to use HTML.
- I changed the titles for the articles at the next-steps section of v3 to sth more descriptive. Our titles are often obscure so using part of the article's description metadata is more clear.

**Comment by siacomuzzi** [comment]
why `User Search Best Practices` is under `search engine v3` (`/users/search/v3/best-practices`)?

**Comment by siacomuzzi** [comment]
also, why `/users/search/v2/query-syntax` and `/users/search/v3/best-practices` are hidden?

**Comment by siacomuzzi** [comment]
image

is this OK? https://auth0.com/docs/users/search/v3#example-request

**Comment by siacomuzzi** [comment]
please move `Searchable Fields` to the right place:

* `v2` has the section in the main doc: https://auth0.com/docs/users/search/v2#searchable-fields
* `v3` has the section in "Query Syntax" doc: https://auth0.com/docs/users/search/v3/query-syntax#searchable-fields

also, there is some places (for example: https://auth0.com/docs/users/search/v3#migrate-from-search-engine-v2-to-v3) where link to `searchable fields` section is broken.

**Comment by siacomuzzi** [comment]
same for "Search using ranges" section:

* `v2`: https://auth0.com/docs/users/search/v2#search-using-ranges
* `v3`: https://auth0.com/docs/users/search/v3/query-syntax#ranges

could you please review all documents (`/users/search/v2`, `/users/search/v3`, `/users/search/v2/query-syntax`, `/users/search/v3/query-syntax`)? Just to make sure we have all sections in the same place

**Pull request merged**